packages/mapviewer/src/modules/map/components/cesium/CesiumMouseTracker.vue

```diff
@@ -26,7 +27,6 @@ const { t } = useI18n()
 
 const is3DReady = computed(() => store.state.cesium.isViewerReady)
 const projection = computed(() => store.state.position.projection)
-
 const dispatcher = { dispatcher: 'CesiumMouseTracker.vue' }
 const getViewer = inject('getViewer', () => {}, true)
```

Comment:
I'd maybe more want to go in the direction where the default value returned here would be `undefined`, and then you can mount the component only when the function returns something relevant.

Or if not applicable to a `v-if`, as you are the "top component" after MapView here, you could do something like

```javascript
const getCesiumViewer = inject('getViewer'), () => undefined, true)

watch(() => getCesiumViewer(), () => /* do something when the viewer changes */)
```

I really don't like, and want to avoid any `setTimeout` use when we can deal with what we want to wait differently

Reply:
I started by creating a watcher on the viewer and making it a reactive components, but after a few tests, I noticed that the `isViewerReady` flag never changed... and so it was never set to false when we unmounted the 3D view. Since this flag is set to true once the viewer is created and ready to be used, switching that flag seems to work without needing to make the whole viewer reactive, and we no longer need to set a timeout.